service/lib/dinstaller/dbus/questions.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "dbus"
+require "dinstaller/dbus/question"
+
+module DInstaller
+  module DBus
+    # This class mimics the basic functionality of ObjectManager. Note that ruby-dbus does not
+    # support ObjectManager yet.
```

Comment:
this is one thing that is not clear for me, even if ruby-dbus does not implement auto export of object manager, you can still implement that interface yourself, not?

And according to https://dbus.freedesktop.org/doc/dbus-specification.html#standard-interfaces-objectmanager method names and interfaces are different.

Reply:
The commet states "mimics the basic functionality of ObjectManager", but not its API. The question is if we want to provide the same API as ObjectManager or not. I didn't do it because we have to change clients anyway if we finally replace *o.o.DInstaller.Questions* by ObjectManager. So, for now, I created the most minimal API I thought.

Reply:
I do not get why we would need to replace it if we change implementation of ObjectManager interface from hand crafted one to one provided by ruby-dbus? It is same interface, so different implementation should not make different results.